src/nbl/video/CVKLogicalDevice.h

```diff
@@ -178,52 +802,382 @@ class CVKLogicalDevice final : public ILogicalDevice
+            vk_dsLayoutBindings[b].descriptorCount = binding->count;
+            vk_dsLayoutBindings[b].stageFlags = static_cast<VkShaderStageFlags>(binding->stageFlags);
+            vk_dsLayoutBindings[b].pImmutableSamplers = nullptr;
+
+            if (binding->samplers)
+            {
+                assert(binding->count <= MAX_SAMPLER_COUNT_PER_BINDING);
+
+                for (uint32_t i = 0u; i < binding->count; ++i)
+                {
+                    if (binding->samplers[i]->getAPIType() != EAT_VULKAN)
+                        continue;
+
+                    vk_samplers[samplerOffset + i] = static_cast<const CVulkanSampler*>(binding->samplers[i].get())->getInternalObject();
+                }
+
+                vk_dsLayoutBindings[b].pImmutableSamplers = vk_samplers + samplerOffset;
+                samplerOffset += binding->count;
+            }
+        }
+
+        VkDescriptorSetLayoutCreateInfo vk_createInfo = { VK_STRUCTURE_TYPE_DESCRIPTOR_SET_LAYOUT_CREATE_INFO };
+        vk_createInfo.pNext = nullptr; // Each pNext member of any structure (including this one) in the pNext chain must be either NULL or a pointer to a valid instance of VkDescriptorSetLayoutBindingFlagsCreateInfo or VkMutableDescriptorTypeCreateInfoVALVE
+        vk_createInfo.flags = 0; // Todo(achal): I would need to create a IDescriptorSetLayout::SCreationParams for this
+        vk_createInfo.bindingCount = bindingCount;
+        vk_createInfo.pBindings = vk_dsLayoutBindings;
+
+        VkDescriptorSetLayout vk_dsLayout;
+        if (vkCreateDescriptorSetLayout(m_vkdev, &vk_createInfo, nullptr, &vk_dsLayout) == VK_SUCCESS)
+        {
+            return core::make_smart_refctd_ptr<CVulkanDescriptorSetLayout>(
+                core::smart_refctd_ptr<CVKLogicalDevice>(this), _begin, _end, vk_dsLayout);
+        }
+        else
+        {
+            return nullptr;
+        }
     }
 
     core::smart_refctd_ptr<IGPUPipelineLayout> createGPUPipelineLayout_impl(const asset::SPushConstantRange* const _pcRangesBegin = nullptr,
-        const asset::SPushConstantRange* const _pcRangesEnd = nullptr, core::smart_refctd_ptr<IGPUDescriptorSetLayout>&& _layout0 = nullptr,
-        core::smart_refctd_ptr<IGPUDescriptorSetLayout>&& _layout1 = nullptr, core::smart_refctd_ptr<IGPUDescriptorSetLayout>&& _layout2 = nullptr,
-        core::smart_refctd_ptr<IGPUDescriptorSetLayout>&& _layout3 = nullptr) override
+        const asset::SPushConstantRange* const _pcRangesEnd = nullptr, core::smart_refctd_ptr<IGPUDescriptorSetLayout>&& layout0 = nullptr,
+        core::smart_refctd_ptr<IGPUDescriptorSetLayout>&& layout1 = nullptr, core::smart_refctd_ptr<IGPUDescriptorSetLayout>&& layout2 = nullptr,
+        core::smart_refctd_ptr<IGPUDescriptorSetLayout>&& layout3 = nullptr) override
     {
-        return nullptr;
+        constexpr uint32_t MAX_PC_RANGE_COUNT = 100u;
+        constexpr uint32_t MAX_DESCRIPTOR_SET_LAYOUT_COUNT = 4u; // temporary max, I believe
```

Comment:
nope, engine mandated Cross-API permament maximum count of descriptor sets

Reply:
Resolved